STAC-22870: add reference docs and guide for otel component and relation mappings (#159)
rancher/stackstate-product-docs: docs/latest/modules/en/pages/setup/otel/sampling.adoc

Tail sampling postpones the sampling decision until a trace is (almost) complete. This allows the tail sampler to make  based on the entire traces, for example to always sample failed traces and/or slow traces. There are many more possibilities of course. The Open Telemetry collector has a https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/tailsamplingprocessor[tail sampling processor] to apply tail sampling.

> So the main advantage of tail sampling is the much bigger flexibility it provides in making sampling. But it comes at a price:

Reviewer: ```suggestion
The advantage of tail sampling is the additional flexibility it provides in making sampling. 
```
